### PR TITLE
Simplify My Traits Components giant refactor

### LINKED ISSUE #465: Simplify My Traits Components
# Tech Debt Request

**As an** engineer
**I would** like to be able to do development work on the product tag, location and roles modal quickly
**so that** I can get features out quicker

Acceptance Criteria

Given that I am working on a feature
When I am working on the MyTraits.tsx component and related react files
Then it is easy to see what is going on and what I need to change

## Notes

The MyTraits file, traits folder, and overall, the tags and roles modals are very confusing to work with and add a lot of extra time onto developing within them.

### PR DESCRIPTION
**Pull request by willot**
## Issue
Resolves #465  

## What was done
- [x] We rebase stage into the branch to make the merge easier and see where thing were diverging 
- [x] Trait component was refactored and rename to the concept of tag(Role, location and product tag) See Ashley for more explanation
## How to test
1. Everything should be working how it was working before

### FOR DEVS
Feature URL: https://featurebranchui.apps.pd01i.edc1.cf.ford.com/

Feature Branch Deployment: https://jenkins-fordlabs.apps.pd01e.edc1.cf.ford.com/job/PeopleMover/job/Deploy%20PeopleMover%20FeatureBranch/

### FOR APPROVAL
Approval URL: https://approvalbranchui.apps.pd01i.edc1.cf.ford.com/

Approval Branch Deployment: https://jenkins-fordlabs.apps.pd01e.edc1.cf.ford.com/job/PeopleMover/job/Deploy%20PeopleMover%20ApprovalBranch/

### CODE REVIEW

File: ui/src/Locations/Location.interface.ts [hunk 1]
```diff
@@ -15,6 +15,6 @@
  * limitations under the License.
  */
 
-import {Trait} from '../Traits/Trait';
+import {Tag} from '../Tags/Tag.interface';
 
-export type SpaceLocation = Trait;
\ No newline at end of file
+export type Location = Tag;
```
Comment (RandomAshes) [inline ui/src/Locations/Location.interface.ts:20]: Can we name this `LocationTag` to match `RoleTag`?

File: ui/src/ModalFormComponents/EditTagRow.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        return tagInputValue.name === ''
+        || showDuplicatedTagErrorMessage
+        || (newNameIsDuplicated(tagInputValue.name) && isColorTheSame());
+    };
+
+    return (
+        <>
+            <div className={`editTagRow ${tagNameClass}`}
+                data-testid={createDataTestId('editTagRow', testIdSuffix)}>
+                { colors && <ColorDropdown selectedColorId={(tagInputValue as RoleEditRequest).colorId} colors={colors} handleColorChange={handleColorChange} />}
```
Comment (RandomAshes) [inline ui/src/ModalFormComponents/EditTagRow.tsx:102]: Can we split this up onto different lines?

File: ui/src/Roles/RoleEditRequest.interface.ts [hunk 3]
```diff
@@ -15,7 +15,8 @@
...
-export interface LocationEditRequest {
-    id: number;
-    updatedName: string;
-}
\ No newline at end of file
+import {TagRequest} from '../Tags/TagRequest.interface';
+
+export interface RoleEditRequest extends TagRequest {
+    colorId?: number;
+}
```
Comment (RandomAshes) [inline ui/src/Roles/RoleEditRequest.interface.ts:22]: Is this necessary if the RoleEditRequest and RoleAddRequest are the same thing?

Comment (willot) [reply ui/src/Roles/RoleEditRequest.interface.ts:22]: no it is not. But we decided to stop and make the objects another refactor

**Review by RandomAshes — APPROVED** [review]
Looks good to me

**Review by smle225 — APPROVED** [review]

**Review by sabdul14 — APPROVED** [review]

**Pull request merged**